### ISSUE TITLE
Move counter-reset outside of mako

### ISSUE THREAD

**Issue by DonatJR** [servo/servo#19387]
Part of #19015. I'd like to work on the -counter-reset longhand.

This one lives in components/style/properties/longhand/counter.mako.rs.

**Comment by DonatJR** [comment]
@emilio I moved the code for counter-reset out of the mako file, but the CounterReset struct has a cfg attribute `#[cfg_attr(feature = "gecko", derive(MallocSizeOf))]` and building the geckolib throws some mismatched type errors.
I noticed there is a `gecko.mako.rs` file as well which still uses the old struct generated by mako (counter-increment and counter-reset shared the `SpecifiedValue` / `ComputedValue` struct, which is why the struct can still be found there). Do I need to move the code in `gecko.mako.rs` as well and if so, where should I put it?
